### ISSUE TITLE
The integration tests are not running

### ISSUE THREAD

**Issue by jorgecarleitao** [apache/arrow-rs#690]
The workflow got broken around June the 30th.

It seems that it stopped running after #439, that broke the workflow.

Unfortunately, github was not very helpful here, as it did not warn us for some reason [on the PR](https://github.com/apache/arrow-rs/pull/439)

**Comment by alamb** [comment]
There are more details on this note on the mailing list too: https://lists.apache.org/thread.html/r84491f36d7e36be1b2e6d21466963ae5401669e84c3a2ea9ef34de55%40%3Cdev.arrow.apache.org%3E

**Comment by jorgecarleitao** [comment]
ahah, you beat me to it. I am still figuring out what happened and will update the description accordingly.

**Comment by kszucs** [comment]
https://github.com/apache/arrow-rs/pull/691 is going to fix it.

**Comment by kszucs** [comment]
There is [a regression](https://github.com/apache/arrow-rs/pull/691/checks?check_run_id=3473126119#step:6:10379) caused by https://github.com/apache/arrow-rs/commit/c87a7bc1e8aa943d883801eaedd1d5eb0bb9312a

**Comment by emkornfield** [comment]
@kszucs will #691 address the integration test failures that are now being seen on "apache/arrow" or should I open a separate issue to track them?

**Comment by kszucs** [comment]
@emkornfield where do we have integration test failures? Could you paste a link?

**Comment by jorgecarleitao** [comment]
@emkornfield @kszucs I think that #691 did solve it, @kszucs applied a fix to the list equality; the integration tests are green on that PR.

**Comment by kszucs** [comment]
Resolved by https://github.com/apache/arrow-rs/commit/34df13a97ce6ae7791599dc2abedf715b79ed690
